Userland/Libraries/LibWeb/Painting/PaintableBox.cpp

```diff
@@ -179,6 +179,10 @@ void PaintableBox::paint_border(PaintContext& context) const
         .bottom = computed_values().border_bottom(),
         .left = computed_values().border_left(),
     };
+
+    if (computed_values().clip().is_rect() && computed_values().position() == CSS::Position::Absolute)
+        context.painter().add_clip_rect(computed_values().clip().to_rect().resolved(Paintable::layout_node(), absolute_border_box_rect()).to_rounded<int>());
+
```

Comment:
Ah another thing, if you add the clip rect to the painter you need to restore it at some point, and you need to save the previous state of the painter. See how overflow clipping works (I think this might just be a special case of that).

Basically for overflow clipping the painter is saved and the clip started in `before_children_paint()`, then
the painter is restored in `after_children_paint()`.

Reply:
Ok this is done.